dashboard/app/assets/javascripts/levels/contract_match.js.jsx

```diff
@@ -0,0 +1,257 @@
+$(window).load(function () {
+
+  /**
+   * @param fn function to be curried with arguments
+   * @params [...] rest of args
+   * @returns {Function} new function with the given arguments pre-set as the
+   *                     defaults (curried)
+   */
+  var curry = function(fn) {
+    var args = Array.prototype.slice.call(arguments, 1);
+
+    return function () {
+      return fn.apply(this, args.concat(
+        Array.prototype.slice.call(arguments, 0)));
+    };
+  };
```

Comment:
Any reason not to use an existing curry function (lodash has one, I bet jquery does) instead of rolling our own?